cpp/src/community/ecg.cu

```diff
@@ -42,25 +43,20 @@ binsearch_maxle(const IndexType *vec, const IndexType val, IndexType low, IndexT
 }
 
 template <typename IdxT, typename ValT>
-__global__ void match_check_kernel(IdxT size,
-                                   IdxT num_verts,
-                                   IdxT *offsets,
-                                   IdxT *indices,
-                                   IdxT *permutation,
-                                   IdxT *parts,
-                                   ValT *weights)
+__global__ void match_check_kernel(
+  IdxT size, IdxT num_verts, IdxT *offsets, IdxT *indices, IdxT *parts, ValT *weights)
```

Comment:
Should we better use thrust?

thrust::transform(
...
[...]__device__(auto ...) {
  auto source = thrust::upper_bound(thrust::seq, ...);
  ...
}
);

What do we get by writing a custom kernel?

Reply:
Don't know.  This custom kernel was already there.  My clang-format restructured this line.

I'd rather address this issue in the MNMG ECG implementation when we refactor this than to do this now.  I'm adding a FIXME to remind me to do that.

Reply:
I agree w/ @seunghwak . The long-held policy has been to avoid custom-written kernels as much as possible. And use thrust / CUB instead.

Reply:
Let me restate my response, perhaps I was too terse.

I agree, this seems like a place where we should just use thrust/cub.  There's no reason that I can think of that this custom kernel would be better than a thrust/cub implementation.  There's nothing in this kernel that takes advantage of features that would make a custom kernel perform better, and generally it increases the maintenance cost of the code.

I modified this file (`ecg.cu`) solely for the purpose of using the Dendrogram class, since the ECG algorithm calls Louvain and I modified Louvain.  I'm reluctant, in general, to rewrite things in the code that I tangentially touch.  It tends to make the work continue without getting anything finished and merged into the baseline.

The only reason this was even in the list of changes is because I ran `clang-format` and it changed the format of these lines of code.  I made no changes relevant to this kernel or the calls to it.

I have a branch where I have started the MNMG ECG work.  In that branch I will completely refactor how the ECG code works.  My intention is to remove this kernel in that branch which hopefully will be merged sometime during release 0.19.

If we think it's critical that I do this for release 0.18, I can certainly take a few hours and refactor this once I'm done addressing the `device_uvector` request.